### PR TITLE
Add support for automatically fixing linting errors with `--fix`

### PR DESCRIPTION
**Pull request by Page-**
Change-type: minor

### CODE REVIEW

File: lib/resin-lint.ts [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import * as Bluebird from 'bluebird';
+// tslint:disable-next-line:no-var-requires
 const coffeelint: any = require('coffeelint');
+// tslint:disable-next-line:no-var-requires
 const reporter: any = require('coffeelint/lib/reporters/default');
 import * as depcheck from 'depcheck';
 import * as fs from 'fs';
 import * as glob from 'glob';
-const merge: any = require('merge');
+import { merge } from 'lodash';
```
Comment (Page-) [inline lib/resin-lint.ts]: I switched to using lodash merge because it has typings and I know it's maintained

File: lib/resin-lint.ts [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import * as Bluebird from 'bluebird';
+// tslint:disable-next-line:no-var-requires
```
Comment (Page-) [inline lib/resin-lint.ts]: Turns out resin-lint didn't lint itself..

File: package.json [hunk 3]
```diff
@@ -10,15 +10,15 @@
...
   },
   "scripts": {
-    "build": "npm run prettify && tsc && npm run lint",
-    "lint": "./bin/resin-lint lib/",
+    "build": "tsc && npm run prettify && npm run lint",
+    "lint": "node ./bin/resin-lint lib/ && node ./bin/resin-lint --typescript lib/",
     "prepublish": "require-npm4-to-publish",
     "prepublishOnly": "npm run test",
-    "prettify": "prettier --config ./config/.prettierrc --write \"lib/**/*.ts\"",
+    "prettify": "node ./bin/resin-lint --typescript --fix lib/",
```
Comment (Page-) [inline package.json]: Switched to using the built-in auto fixing, it essentially acts as a test in itself

File: package.json [hunk 4]
```diff
@@ -10,15 +10,15 @@
     "url": "git@github.com:resin-io/node-resin-lint"
   },
   "scripts": {
-    "build": "npm run prettify && tsc && npm run lint",
-    "lint": "./bin/resin-lint lib/",
+    "build": "tsc && npm run prettify && npm run lint",
+    "lint": "node ./bin/resin-lint lib/ && node ./bin/resin-lint --typescript lib/",
```
Comment (Page-) [inline package.json]: Using `node` explicitly makes life easier on windows

File: lib/resin-lint.ts [hunk 5]
```diff
@@ -256,6 +268,7 @@ export const lint = (passedParams: any) =>
 				const prettierCheck = options.argv.prettier !== false;
 				const testsCheck = options.argv.tests === true;
 				const typescriptCheck = options.argv.typescript;
+				const autoFix = options.argv.fix;
```
Comment (dfunckt) [inline lib/resin-lint.ts]:
```suggestion
				const autoFix = options.argv.fix === true;
```

like above. In fact, I wonder what `optimist` does for non-provided boolean options -- is the option `undefined`?

**Pull request merged**